R/module_init_data.R

```diff
@@ -80,41 +80,48 @@ srv_init_data <- function(id, data, modules, filter = teal_slices()) {
     } else if (inherits(data, "teal_data")) {
       reactiveVal(data)
     } else if (test_reactive(data)) {
-      .fallback_on_failure(this = data, that = reactive(req(FALSE)), label = "Reactive data")
+      data
+    } else if (inherits(data, "qenv.error")) {
+      reactive(data)
```

Comment:
```suggestion
```

Reply:
In a Teal Module, even if `teal_data` is the input source it can be created inside the shiny server, so the shiny app is already running. It's just the perspective of whether we want to disconnect the app when there is `qenv.error`.

If we handle `qenv.error` inputs this app will not disconnect and there will be an error message displayed similar to the error that happens during `teal_data_module` qenv error.
Do you prefer to have a disconnection or UI error for this case (example app below)?

```r
modules <- modules(example_module(), example_module("mtcars only", datanames = "mtcars"))
ui <- fluidPage(
  "Custom UI",
  ui_teal(id = "teal_1", modules = modules),
  ui_teal(id = "teal_2", modules = modules)
)
server <- function(input, output, session) {
  data <- teal_data(iris = iris, mtcars = mtcars)
  srv_teal(
    id = "teal_1",
    data = within(teal_data(), {
      stop("error")
    }),
    modules = modules
  )
  srv_teal(
    id = "teal_2",
    data = within(teal_data(), {
      stop("error")
    }),
    modules = modules
  )
}
shinyApp(ui, server)
```

Reply:
I think thats debatale. I prefer to disconnect to immediately inform app-developer that there is no way that this app can work. I agree that if reactive returns error then we should do everything to keep an app running, but not on initialize - `qenv.error` is an initialization error.

Reply:
Got it, this makes sense. Let me remove it.